youtube_dl/extractor/markiza.py

```diff
@@ -26,54 +28,89 @@ class MarkizaIE(InfoExtractor):
-        'url': 'http://videoarchiv.markiza.sk/video/televizne-noviny/televizne-noviny/85430_televizne-noviny',
+        'url': ' https://videoarchiv.markiza.sk/video/laska-na-prenajom/epizoda/58779-seria-1-epizoda-14',
         'info_dict': {
             'id': '85430',
             'title': 'Televízne noviny',
         },
         'playlist_count': 23,
     }, {
-        'url': 'http://videoarchiv.markiza.sk/video/oteckovia/84723',
+        'url': 'https://videoarchiv.markiza.sk/video/oteckovia/84723',
         'only_matching': True,
     }, {
-        'url': 'http://videoarchiv.markiza.sk/video/84723',
+        'url': 'https://videoarchiv.markiza.sk/video/84723',
         'only_matching': True,
     }, {
-        'url': 'http://videoarchiv.markiza.sk/video/filmy/85190_kamenak',
+        'url': 'https://videoarchiv.markiza.sk/video/filmy/85190_kamenak',
         'only_matching': True,
     }, {
-        'url': 'http://videoarchiv.markiza.sk/video/reflex/zo-zakulisia/84651_pribeh-alzbetky',
+        'url': 'https://videoarchiv.markiza.sk/video/reflex/zo-zakulisia/84651_pribeh-alzbetky',
         'only_matching': True,
     }, {
-        'url': 'http://videoarchiv.markiza.sk/embed/85295',
+        'url': 'https://videoarchiv.markiza.sk/embed/85295',
         'only_matching': True,
     }]
 
     def _real_extract(self, url):
         video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+        embed = self._search_regex(
+            r'<iframe src="(https:\/\/media.*?)" style="',
+            webpage, 'embed', fatal=False)
+        webpage = self._download_webpage(embed, video_id)
+        data = self._search_regex(
+            r'processAdTagModifier\((\{.*)\), {"video":',
+            webpage, 'embed', fatal=False)
+        data2 = self._search_regex(
+            r'processAdTagModifier\(\{.*\), ({"video":.*)\);',
+            webpage, 'embed', fatal=False)
+        data = re.sub('\\\\/', '/', data)
+        data2 = re.sub('\\\\/', '/', data2)
+        info = json.loads(data)
+        info2 = json.loads(data2)
 
-        data = self._download_json(
-            'http://videoarchiv.markiza.sk/json/video_jwplayer7.json',
-            video_id, query={'id': video_id})
-
-        info = self._parse_jwplayer_data(data, m3u8_id='hls', mpd_id='dash')
-
-        if info.get('_type') == 'playlist':
-            info.update({
-                'id': video_id,
-                'title': try_get(
-                    data, lambda x: x['details']['name'], compat_str),
-            })
-        else:
-            info['duration'] = parse_duration(
-                try_get(data, lambda x: x['details']['duration'], compat_str))
-        return info
+        formats = []
+        for format_id, format_list in info['tracks'].items():
+            if not isinstance(format_list, list):
+                format_list = [format_list]
+            for format_dict in format_list:
+                if not isinstance(format_dict, dict):
```

Comment:
Here (ll.77-9) you can use `try_get()` to simplify and enforce the `format_url`:
```
                format_url = try_get(format_dict, lambda x: url_or_none(x['src']))
                if not format_url:
                    continue
                # now guaranteed: format_dict is a dict-like object, format_url is not None
```